invenio_rdm_records/resources/serializers.py

```diff
@@ -21,87 +20,60 @@ class UIJSONSerializer(JSONSerializer):
-
-        :params labels_map: Dict of transforming bucket values to labels.
-        :params bucket: Current aggregation bucket.
-        """
-        bucket_label = labels_map.get(bucket["key"])
-        if bucket_label:
-            bucket["label"] = bucket_label
-        if bucket.get("subtype"):
-            bucket["subtype"]["buckets"] = list(map(
-                partial(self._set_bucket_label, labels_map),
-                bucket["subtype"]["buckets"]))
-        return bucket
-
-    def _set_buckets_labels(self, label_map, buckets):
-        """Map buckets values to labels according to configuration.
-
-        :params labels_map: Dict of transforming bucket values to labels.
-        :params buckets: Current aggregation bucket list.
-        """
-        return list(map(
-                partial(self._set_bucket_label, label_map),
-                buckets))
-
-    def _serialize_agg(self, agg_obj, labels_map):
-        """Inject ui config for aggregation.
+    def _enrich_buckets_with_labels(self, agg_obj, vocabulary):
+        """Inject labels in the aggregation buckets.
 
         :params agg_obj: Current aggregation object.
-        :params labels_map: Dict of transforming bucket values to labels.
+        :params vocabulary: Dict with vocabulary data.
         """
-        if agg_obj:
-            agg_obj["buckets"] = self._set_buckets_labels(
-                labels_map, agg_obj["buckets"]
-            )
-
-    def _serialize_resource_type_agg(self, resource_type_agg):
-        """Inject ui config for `resource_type` aggregation."""
-        resource_type_labels = self._serialize_ui_options_from_vocabulary(
-            "resource_type", "type")
-        self._serialize_agg(resource_type_agg, resource_type_labels["type"])
-
-    def _serialize_access_right_agg(self, access_right_agg):
-        """Inject ui config for `access_right` aggregation."""
-        access_right_labels = self._serialize_ui_options_from_vocabulary(
-            "access_right")
-        self._serialize_agg(access_right_agg, access_right_labels)
+        buckets = agg_obj['buckets']
+        for bucket in buckets:
+            for k in list(bucket):
+                if k == 'key':
+                    bucket['label'] = str(vocabulary.get(bucket[k])["text"])
+                if type(bucket[k]) is dict and 'buckets' in bucket[k]:
+                    self._enrich_buckets_with_labels(bucket[k], vocabulary)
 
     def _serialize_aggregations(self, obj_list):
         """Inject ui config in aggregations."""
         aggregations = obj_list.get("aggregations")
         if aggregations:
-            self._serialize_access_right_agg(aggregations.get("access_right"))
-            self._serialize_resource_type_agg(
-                aggregations.get("resource_type"))
+            for aggregation_key in aggregations:
+                agg_vocabulary = self._serialize_ui_options_from_vocabulary(
+                    aggregation_key)
+                if not agg_vocabulary:
```

Comment:
If `agg_vocabulary` value is actually `None`, maybe is better to rewrite this into `if agg_vocabulary is None` :)